File: backend/seed.py

import json
from app import app, db
from models import User, Task, Badge, QuizQuestion
from werkzeug.security import generate_password_hash

def seed_database():
    print("Seeding database...")
    
    # Create tasks
    tasks = [
        Task(text="Use a reusable bottle", description="Avoid single-use plastic bottles", points=10),
        Task(text="Switch off unused lights", description="Conserve energy throughout the day", points=10),
        Task(text="Take a shorter shower", description="Conserve water by reducing shower time", points=10),
        Task(text="Use public transportation or walk", description="Reduce carbon emissions from private vehicles", points=10),
        Task(text="Eat a plant-based meal", description="Reduce environmental impact of meat consumption", points=10),
        Task(text="Avoid single-use plastics today", description="Decline plastic straws, bags, and containers", points=10),
    ]
    
    # Create badges
    badges = [
        Badge(name="Green Starter", description="Completed your first eco-friendly task", icon="🌱", points_required=10),
        Badge(name="Eco Enthusiast", description="Earned 50 points through eco-friendly actions", icon="🌿", points_required=50),
        Badge(name="Planet Protector", description="Earned 100 points by being environmentally conscious", icon="🌍", points_required=100),
        Badge(name="Sustainability Champion", description="Reached 200 points with your green habits", icon="🏆", points_required=200),
        Badge(name="Climate Hero", description="An impressive 500 points earned through eco-actions", icon="⭐", points_required=500),
    ]
    
    # Create quiz questions
    questions = [
        QuizQuestion(
            text="Which of these actions saves the most water?",
            options=json.dumps([
                "Taking shorter showers", 
                "Using a dishwasher instead of washing by hand", 
                "Turning off the tap while brushing teeth", 
                "Washing clothes in cold water"
            ]),
            correct_answer=1
        ),
        QuizQuestion(
            text="What is the most effective way to reduce your carbon footprint?",
            options=json.dumps([
                "Using public transportation", 
                "Eating less meat", 
                "Recycling regularly", 
                "Using energy-efficient appliances"
            ]),
            correct_answer=1
        ),
        QuizQuestion(
            text="Which material takes the longest to decompose in a landfill?",
            options=json.dumps([
                "Paper", 
                "Aluminum cans", 
                "Plastic bottles", 
                "Glass bottles"
            ]),
            correct_answer=2
        ),
        QuizQuestion(
            text="What percentage of the Earth's surface is covered by water?",
            options=json.dumps([
                "50%", 
                "60%", 
                "70%", 
                "80%"
            ]),
            correct_answer=2
        ),
        QuizQuestion(
            text="Which of these is NOT a renewable resource?",
            options=json.dumps([
                "Solar energy", 
                "Wind power", 
                "Natural gas", 
                "Hydroelectric power"
            ]),
            correct_answer=2
        ),
    ]
    
    # Add demo user
    demo_user = User(
        username="demo_user",
        email="demo@example.com",
        password_hash=generate_password_hash("password123"),
        points=0
    )
    
    # Add everything to the database
    db.session.add(demo_user)
    
    for task in tasks:
        db.session.add(task)
    
    for badge in badges:
        db.session.add(badge)
    
    for question in questions:
        db.session.add(question)
    
    db.session.commit()
    print("Database seeded successfully!")

if __name__ == '__main__':
    with app.app_context():
        db.create_all()
        seed_database()

File: backend-nodejs/seed.js

const mongoose = require('mongoose');
require('dotenv').config();

const Task = require('./models/Task');
const Question = require('./models/Question');
const Badge = require('./models/Badge');

// Connect to MongoDB
mongoose.connect(process.env.MONGODB_URI || 'mongodb://localhost:27017/ecowatt')
  .then(() => console.log('Connected to MongoDB for seeding'))
  .catch(err => console.error('MongoDB connection error:', err));

const seedTasks = [
  {
    text: "Use a reusable bottle",
    description: "Replace single-use plastic bottles with a reusable water bottle",
    category: "waste",
    points: 10,
    icon: "💧"
  },
  {
    text: "Switch off unused lights",
    description: "Turn off lights when leaving a room to save energy",
    category: "energy",
    points: 5,
    icon: "💡"
  },
  {
    text: "Take a shorter shower",
    description: "Reduce shower time by 2 minutes to conserve water",
    category: "water",
    points: 8,
    icon: "🚿"
  },
  {
    text: "Use public transportation or walk",
    description: "Choose eco-friendly transportation options",
    category: "transport",
    points: 15,
    icon: "🚌"
  },
  {
    text: "Eat a plant-based meal",
    description: "Choose a vegetarian or vegan meal today",
    category: "food",
    points: 12,
    icon: "🥗"
  },
  {
    text: "Avoid single-use plastics today",
    description: "Say no to plastic bags, straws, and containers",
    category: "waste",
    points: 10,
    icon: "🌱"
  }
];

const seedQuestions = [
  {
    text: "What percentage of the Earth's surface is covered by water?",
    options: ["50%", "60%", "71%", "80%"],
    correctAnswer: 2,
    category: "water",
    difficulty: "easy",
    points: 5
  },
  {
    text: "Which greenhouse gas is most abundant in Earth's atmosphere?",
    options: ["Carbon dioxide", "Methane", "Water vapor", "Nitrous oxide"],
    correctAnswer: 2,
    category: "climate",
    difficulty: "medium",
    points: 10
  },
  {
    text: "How long does it take for a plastic bottle to decompose?",
    options: ["10 years", "50 years", "100 years", "450+ years"],
    correctAnswer: 3,
    category: "waste",
    difficulty: "medium",
    points: 10
  },
  {
    text: "What is the most effective way to reduce your carbon footprint?",
    options: ["Recycling", "Using LED bulbs", "Reducing meat consumption", "Taking shorter showers"],
    correctAnswer: 2,
    category: "climate",
    difficulty: "hard",
    points: 15
  },
  {
    text: "Which renewable energy source generates the most electricity globally?",
    options: ["Solar", "Wind", "Hydroelectric", "Geothermal"],
    correctAnswer: 2,
    category: "energy",
    difficulty: "medium",
    points: 10
  }
];

const seedBadges = [
  {
    name: "Eco Starter",
    description: "Complete your first eco-friendly task",
    icon: "🌱",
    pointsRequired: 10,
    category: "tasks"
  },
  {
    name: "Green Warrior",
    description: "Earn 50 points through eco-friendly actions",
    icon: "⚔️",
    pointsRequired: 50,
    category: "tasks"
  },
  {
    name: "Quiz Master",
    description: "Answer 10 quiz questions correctly",
    icon: "🧠",
    pointsRequired: 50,
    category: "quiz"
  },
  {
    name: "Eco Champion",
    description: "Reach 100 points and become an environmental champion",
    icon: "🏆",
    pointsRequired: 100,
    category: "special"
  },
  {
    name: "Planet Protector",
    description: "Earn 200 points protecting our planet",
    icon: "🌍",
    pointsRequired: 200,
    category: "special"
  }
];

const seedDatabase = async () => {
  try {
    // Clear existing data
    await Task.deleteMany({});
    await Question.deleteMany({});
    await Badge.deleteMany({});

    // Seed tasks
    await Task.insertMany(seedTasks);
    console.log('✅ Tasks seeded successfully');

    // Seed questions
    await Question.insertMany(seedQuestions);
    console.log('✅ Quiz questions seeded successfully');

    // Seed badges
    await Badge.insertMany(seedBadges);
    console.log('✅ Badges seeded successfully');

    console.log('🎉 Database seeding completed successfully!');
    process.exit(0);
  } catch (error) {
    console.error('❌ Error seeding database:', error);
    process.exit(1);
  }
};

seedDatabase();
